src/context/UserContext.tsx

```diff
@@ -0,0 +1,97 @@
+// This is taken from https://github.com/supabase-community/auth-ui/blob/main/packages/react/src/components/Auth/UserContext.tsx
+//  Then updated so it works with the older version of Supabase JS
+
+import { useEffect, useState, createContext, useContext } from 'react';
+import { Session, User } from '@supabase/supabase-js';
+import { useClient } from 'hooks/supabase-swr';
+import { BaseComponentProps } from 'types';
+import { logRocketConsole, logRocketEvent } from 'services/shared';
+import { CustomEvents } from 'services/types';
+
+export interface AuthSession {
+    user: User | null;
+    session: Session | null;
+}
+
+const UserContext = createContext<AuthSession | undefined>({
+    user: null,
+    session: null,
+});
+
+const UserContextProvider = ({ children }: BaseComponentProps) => {
+    const supabaseClient = useClient();
+    const [session, setSession] = useState<Session | null>(null);
+    const [user, setUser] = useState<User | null>(session?.user ?? null);
+
+    useEffect(() => {
+        void (async () => {
+            const authSession = supabaseClient.auth.session();
+
+            // If there is no session set to null and logout
+            if (!authSession) {
+                setSession(null);
+                setUser(null);
+                logRocketEvent(CustomEvents.AUTH_SIGNOUT, {
+                    trigger: 'UserContext:session',
+                });
+                await supabaseClient.auth.signOut();
+                return;
+            }
+
+            // Set the latest session into state
+            setSession(authSession);
+
+            // Fetch the user info from the server to ensure we know who the user truly is
+            const { data } = await supabaseClient.auth.api.getUser(
+                authSession.access_token
+            );
```

Comment:
This is a bigger change. We now get user data directly from the server instead of the local storage. Even though we don't really ever use this info for security purposes this felt a little safer.